jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/config/FastJson1IT.java

```diff
@@ -0,0 +1,115 @@
+        assertThat(schemaRule.getGenerateDir(), not(containsText("org.codehaus.jackson")));
+        assertThat(schemaRule.getGenerateDir(), not(containsText("com.fasterxml.jackson")));
+        assertThat(schemaRule.getGenerateDir(), not(containsText("com.google.gson")));
+        assertThat(schemaRule.getGenerateDir(), not(containsText("@SerializedName")));
+        assertThat(schemaRule.getGenerateDir(), not(containsText("com.squareup.moshi")));
+        assertThat(schemaRule.getGenerateDir(), not(containsText("@com.squareup.moshi.Json")));
+        assertThat(schemaRule.getGenerateDir(), containsText("com.alibaba.fastjson"));
+        assertThat(schemaRule.getGenerateDir(), containsText("@JSONField"));
+
+        Method getter = generatedType.getMethod("getBuild");
+
+        assertThat(generatedType.getAnnotation(JsonPropertyOrder.class), is(nullValue()));
+        assertThat(generatedType.getAnnotation(JsonInclude.class), is(nullValue()));
+        assertThat(getter.getAnnotation(JsonProperty.class), is(nullValue()));
+    }
+
+    @Test
+    public void annotationStyleFastJsonMakesTypesThatWorkWithFastJson() throws ClassNotFoundException, SecurityException, IOException {
+
+        ClassLoader resultsClassLoader = schemaRule.generateAndCompile("/json/examples/", "com.example",
+                config("annotationStyle", "fastjson",
+                        "propertyWordDelimiters", "_",
+                        "sourceType", "json",
+                        "useLongIntegers", true));
+
+        assertJsonRoundTrip(resultsClassLoader, "com.example.Torrent", "/json/examples/torrent.json");
+        assertJsonRoundTrip(resultsClassLoader, "com.example.GetUserData", "/json/examples/GetUserData.json");
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    @Test
+    public void enumValuesAreSerializedCorrectly() throws ClassNotFoundException, NoSuchMethodException, SecurityException, InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+
+        ClassLoader resultsClassLoader = schemaRule.generateAndCompile("/schema/enum/typeWithEnumProperty.json", "com.example",
+                config("annotationStyle", "fastjson",
+                        "propertyWordDelimiters", "_"));
+
+        Class generatedType = resultsClassLoader.loadClass("com.example.TypeWithEnumProperty");
+        Class enumType = resultsClassLoader.loadClass("com.example.TypeWithEnumProperty$EnumProperty");
+        Object instance = generatedType.newInstance();
+
+        Method setter = generatedType.getMethod("setEnumProperty", enumType);
+        setter.invoke(instance, enumType.getEnumConstants()[3]);
+
+        String json = JSON.toJSONString(instance);
+        Map<String, String> jsonAsMap = JSON.parseObject(json, Map.class);
+
+        assertThat(jsonAsMap.get("enum_Property"), is("4 ! 1"));
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
```

Comment:
`@SuppressWarnings({"rawtypes", "unchecked"})` could be removed if `Class generatedType` would be replaced with `Class<?> generatedType`